sdks/java/extensions/avro/src/main/java/org/apache/beam/sdk/extensions/avro/schemas/utils/AvroUtils.java

```diff
@@ -1186,6 +1196,9 @@ private static org.apache.avro.Schema getFieldSchema(
         } else if (SqlTypes.TIMESTAMP.getIdentifier().equals(identifier)) {
           baseType =
               LogicalTypes.timestampMicros().addToSchema(org.apache.avro.Schema.create(Type.LONG));
+        } else if (Timestamp.IDENTIFIER.equals(identifier)) {
+          baseType = org.apache.avro.Schema.create(Type.LONG);
+          baseType.addProp("logicalType", TIMESTAMP_NANOS_LOGICAL_TYPE);
```

Comment:
`Timestamp.IDENTIFIER.equals(identifier)` just tells us that it's the Timestamp logical type, but we don't know the precision yet.
We'd need to check the logical type instance's precision, i.e. `logicalType.getArgument()`

This could end up being micros or millis